js/zb.js

```diff
@@ -1185,17 +1261,40 @@ module.exports = class zb extends Exchange {
     }
 
     sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
-        let url = this.urls['api'][api];
-        if (api === 'public') {
-            url += '/' + this.version + '/' + path;
+        const [ section, version, access ] = api;
+        let url = this.urls['api'][section][version][access];
+        if (path === 'getFeeInfo') {
+            url += '/' + path;
             if (Object.keys (params).length) {
                 url += '?' + this.urlencode (params);
             }
-        } else if (api === 'trade') {
-            url += '/' + path;
+        } else if (section === 'spot' && version === 'v1' && access === 'public') {
+            url += '/' + version + '/' + path;
+            if (Object.keys (params).length) {
+                url += '?' + this.urlencode (params);
+            }
+        } else if (section === 'contract' && version === 'v1' && access === 'public') {
+            url += '/' + version + '/' + path;
+            if (Object.keys (params).length) {
+                url += '?' + this.urlencode (params);
+            }
+        } else if (section === 'contract' && version === 'v2' && access === 'public') {
+            url += '/' + version + '/' + path;
             if (Object.keys (params).length) {
                 url += '?' + this.urlencode (params);
             }
+        } else if (section === 'contract' && version === 'v2' && access === 'private') {
```

Comment:
We don't have to do these multiple checks over and over, let's check each thing just one:

```JavaScript
const [ section, version, access ] = api;
if (path === '...') {
    // ...
} else if (section === 'spot') {
    if (access == 'public') {
        // ...
    } else {
        // ...
    }
} else if (section === 'swap') {
    if (version === 'v2') {
        if  (access == 'public') {
            // ...
        } else {
            // private
        }
    } else {
        // v1
    }
}
```

Reply:
Also, a quick question – is it possible that ZB adds USDT-margined futures contracts alongside the swap contracts in the future? If that is the case, we should rename `swap` > `contract` then. It depends on whether their derivatives API can theoretically support expiring futures or just perpetual swaps.

Reply:
I've renamed swap to contract. A lot of the requests have a futuresAccountType parameter so that leads me to believe that they will add additional support to this derivatives API. 

futuresAccountType | Yes | Integer | 1:USDT-M perpetual futures
-- | -- | -- | --

Reply:
@Dan-krm Thanks!